### ISSUE TITLE
spellchecker plugin doesn't work

### ISSUE THREAD

**Issue by TinyMCE-User** [tinymce/tinymce#1672]
The latest version of the spellchecker plugin doesn't work. Attempting to do a spell checking out-of-the-box will yield a {"error":"Missing input parameter 'text'."} error. One of the problems is that the minified javascript is sending json-encoded data. eg.

{"id":"c0","method":"spellcheck","params":{"lang":"en","words":["tgis","is","misspelled","text"]}}

The code, in contrast, is not expecting minified data. From spellchecker\includes\Engine.php:

    $method = self::getParam("method", "spellcheck");
    $lang = self::getParam("lang", "en_US");
    $text = self::getParam("text");

Replacing that with this got me past that error:

    global $HTTP_RAW_POST_DATA;
    $data = json_decode($HTTP_RAW_POST_DATA);
    $method = $data->method;
    $lang = $data->params->lang;
    $text = $data->params->words;

Not that that's the only problem. Since $data->params->words contains an array of words $text, in the above, should be renamed to $words and the check for $text should be removed. But even after that, although EnchantEngine.php correctly finds the misspelled words, it doesn't return them in a format supported by the minified js.

Basically, the plugin doesn't work. I'd fix it myself were it not for the minified js. And my willingness aside I shouldn't have to fix it anyway as it ought to be working outside of the box.

> _**Legacy information imported from TinyMCE bug tracker:**_
>
> _[#T6925](http://archive.tinymce.com/develop/bugtracker_view.php?id=6925) posted by [yawnmoth](http://archive.tinymce.com/profile.php?name=yawnmoth)_

> _**Tags:** [firefox msie safari chrome opera spellchecker]_
> _**Status:** Closed_
> _**Resolution:** Invalid_
> _**Attached URL:** http://www.frostjedi.com/terra/dev/tinymce/test.html_
